f2py build hangs after running gcc w/1.15.0

Sorry for a rather incomplete bug report, but isolating this to a code fragment may not be possible. I am happy to provide more help in diagnosis, though. 

I have an automated Jenkins build procedure that uses f2py to generate .so files from a series of Fortran files. With anaconda packages 

    numpy-1.14.5               |   py27h1b885b7_4          35 KB
    numpy-base-1.14.5          |   py27hdbf6ddf_4         4.1 MB

running on VM with Red Hat Enterprise Linux Server release 7.5 (Maipo), my build runs fine. If I update to 

    numpy                     1.15.0           py27h1b885b7_0  
    numpy-base                1.15.0           py27h3dfced4_0  

The build hangs in this command:

```
jenkins  23248 23087  0 12:10 ?        00:00:02 /var/lib/jenkins/jobs/Py2_L64_Build-copy/workspace/conda27/bin/python /var/lib/jenkins/jobs/Py2_L64_Build-copy/workspace/conda27/bin/f2py -c fellipse.for -m fellipse libspsubs.a libtexturesubs.a libDIFFaXsubs.a libpowsubs.a --fcompiler=gnu95 --f77exec=gfortran --f77flags=-fno-range-check -m64
```
with this following output:

```
building 'fellipse' extension
compiling C sources
C compiler: gcc -pthread -B /var/lib/jenkins/jobs/Py2_L64_Build-copy/workspace/conda27/compiler_compat -Wl,--sysroot=/ -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC

creating /tmp/tmpnkv0Hd/tmp
creating /tmp/tmpnkv0Hd/tmp/tmpnkv0Hd
creating /tmp/tmpnkv0Hd/tmp/tmpnkv0Hd/src.linux-x86_64-2.7
compile options: '-I/tmp/tmpnkv0Hd/src.linux-x86_64-2.7 -I/var/lib/jenkins/jobs/Py2_L64_Build-copy/workspace/conda27/lib/python2.7/site-packages/numpy/core/include -I/var/lib/jenkins/jobs/Py2_L64_Build-copy/workspace/conda27/include/python2.7 -c'
gcc: /tmp/tmpnkv0Hd/src.linux-x86_64-2.7/fellipsemodule.c
gcc: /tmp/tmpnkv0Hd/src.linux-x86_64-2.7/fortranobject.c
```

It apparently completes the gcc step and then hangs. It is not in an infinite loop, since the CPU time is not increasing. This is problem reproducible and restoring the previous numpy version addresses it.

@mattip I will try to look using GDB later tonight and will update you with what I find.

Here is the output of running the same command using both 1.14.5 and 1.15.0:

### 1.14.5
```
running build
running config_cc
unifing config_cc, config, build_clib, build_ext, build commands --compiler options
running config_fc
unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options
running build_src
build_src
building extension "libconfig" sources
f2py options: ['skip:', ':']
f2py:> /tmp/tmpIySmCL/src.linux-x86_64-2.7/libconfigmodule.c
creating /tmp/tmpIySmCL/src.linux-x86_64-2.7
Reading fortran codes...
    Reading file 'config/config.f90' (format:free)
Line #9 in config/config.f90:"    include "config.h""
    readfortrancode: could not find include file 'config.h' in config. Ignoring.
Post-processing...
    Block: libconfig
            Block: config
Post-processing (stage 2)...
    Block: libconfig
        Block: unknown_interface
            Block: config
Building modules...
    Building module "libconfig"...
        Constructing F90 module support for "config"...
          Variables: bin_path lib_path inc_path src_path ancildat_path install_path
    Wrote C/API module "libconfig" to file "/tmp/tmpIySmCL/src.linux-x86_64-2.7/libconfigmodule.c"
    Fortran 90 wrappers are saved to "/tmp/tmpIySmCL/src.linux-x86_64-2.7/libconfig-f2pywrappers2.f90"
  adding '/tmp/tmpIySmCL/src.linux-x86_64-2.7/fortranobject.c' to sources.
  adding '/tmp/tmpIySmCL/src.linux-x86_64-2.7' to include_dirs.
copying /build/usr/local/lib/python2.7/site-packages/numpy/f2py/src/fortranobject.c -> /tmp/tmpIySmCL/src.linux-x86_64-2.7
copying /build/usr/local/lib/python2.7/site-packages/numpy/f2py/src/fortranobject.h -> /tmp/tmpIySmCL/src.linux-x86_64-2.7
  adding '/tmp/tmpIySmCL/src.linux-x86_64-2.7/libconfig-f2pywrappers2.f90' to sources.
 - nothing done with h_files = ['config.h']
build_src: building npy-pkg config files
running build_ext
customize UnixCCompiler
customize UnixCCompiler using build_ext
customize Gnu95FCompiler
Found executable /usr/bin/gfortran
customize Gnu95FCompiler using build_ext
building 'libconfig' extension
compiling C sources
C compiler: gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC

creating /tmp/tmpIySmCL/tmp
creating /tmp/tmpIySmCL/tmp/tmpIySmCL
creating /tmp/tmpIySmCL/tmp/tmpIySmCL/src.linux-x86_64-2.7
compile options: '-I. -I/build/dev_geoips/link/geoips/geoips/geoalgs/include -I/usr/include -I/tmp/tmpIySmCL/src.linux-x86_64-2.7 -I/build/usr/local/lib/python2.7/site-packages/numpy/core/include -I/build/usr/local/include/python2.7 -c'
gcc: /tmp/tmpIySmCL/src.linux-x86_64-2.7/libconfigmodule.c
In file included from /build/usr/local/lib/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1816:0,
                 from /build/usr/local/lib/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,
                 from /build/usr/local/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from /tmp/tmpIySmCL/src.linux-x86_64-2.7/fortranobject.h:13,
                 from /tmp/tmpIySmCL/src.linux-x86_64-2.7/libconfigmodule.c:15:
/build/usr/local/lib/python2.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^
gcc: /tmp/tmpIySmCL/src.linux-x86_64-2.7/fortranobject.c
In file included from /build/usr/local/lib/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1816:0,
                 from /build/usr/local/lib/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,
                 from /build/usr/local/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from /tmp/tmpIySmCL/src.linux-x86_64-2.7/fortranobject.h:13,
                 from /tmp/tmpIySmCL/src.linux-x86_64-2.7/fortranobject.c:2:
/build/usr/local/lib/python2.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^
compiling Fortran 90 module sources
Fortran f77 compiler: /usr/bin/gfortran -Wall -g -ffixed-form -fno-second-underscore -fPIC -O3
Fortran f90 compiler: /usr/bin/gfortran -Wall -g -fno-second-underscore -fPIC -O3
Fortran fix compiler: /usr/bin/gfortran -Wall -g -ffixed-form -fno-second-underscore -Wall -g -fno-second-underscore -fPIC -O3
creating /tmp/tmpIySmCL/config
compile options: '-I. -I/build/dev_geoips/link/geoips/geoips/geoalgs/include -I/usr/include -I/tmp/tmpIySmCL/src.linux-x86_64-2.7 -I/build/usr/local/lib/python2.7/site-packages/numpy/core/include -I/build/usr/local/include/python2.7 -c'
extra options: '-J/tmp/tmpIySmCL/ -I/tmp/tmpIySmCL/'
gfortran:f90: config/config.f90
compiling Fortran sources
Fortran f77 compiler: /usr/bin/gfortran -Wall -g -ffixed-form -fno-second-underscore -fPIC -O3
Fortran f90 compiler: /usr/bin/gfortran -Wall -g -fno-second-underscore -fPIC -O3
Fortran fix compiler: /usr/bin/gfortran -Wall -g -ffixed-form -fno-second-underscore -Wall -g -fno-second-underscore -fPIC -O3
compile options: '-I. -I/build/dev_geoips/link/geoips/geoips/geoalgs/include -I/usr/include -I/tmp/tmpIySmCL/src.linux-x86_64-2.7 -I/build/usr/local/lib/python2.7/site-packages/numpy/core/include -I/build/usr/local/include/python2.7 -c'
extra options: '-J/tmp/tmpIySmCL/ -I/tmp/tmpIySmCL/'
gfortran:f90: /tmp/tmpIySmCL/src.linux-x86_64-2.7/libconfig-f2pywrappers2.f90
/usr/bin/gfortran -Wall -g -Wall -g -shared /tmp/tmpIySmCL/tmp/tmpIySmCL/src.linux-x86_64-2.7/libconfigmodule.o /tmp/tmpIySmCL/tmp/tmpIySmCL/src.linux-x86_64-2.7/fortranobject.o /tmp/tmpIySmCL/config/config.o /tmp/tmpIySmCL/tmp/tmpIySmCL/src.linux-x86_64-2.7/libconfig-f2pywrappers2.o -L. -L/usr/lib -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5 -L/usr/lib/gcc/x86_64-redhat-linux/4.8.5 -L/build/usr/local/lib -lpython2.7 -lgfortran -o ./libconfig.so
Removing build directory /tmp/tmpIySmCL
```
### 1.15.0

```
running build
running config_cc
unifing config_cc, config, build_clib, build_ext, build commands --compiler options
running config_fc
unifing config_fc, config, build_clib, build_ext, build commands --fcompiler options
running build_src
build_src
building extension "libconfig" sources
f2py options: ['skip:', ':']
f2py:> /tmp/tmp0aQ2vk/src.linux-x86_64-2.7/libconfigmodule.c
creating /tmp/tmp0aQ2vk/src.linux-x86_64-2.7
Reading fortran codes...
    Reading file 'config/config.f90' (format:free)
Line #9 in config/config.f90:"    include "config.h""
    readfortrancode: could not find include file 'config.h' in config. Ignoring.
Post-processing...
    Block: libconfig
            Block: config
Post-processing (stage 2)...
    Block: libconfig
        Block: unknown_interface
            Block: config
Building modules...
    Building module "libconfig"...
        Constructing F90 module support for "config"...
          Variables: bin_path lib_path inc_path src_path ancildat_path install_path
    Wrote C/API module "libconfig" to file "/tmp/tmp0aQ2vk/src.linux-x86_64-2.7/libconfigmodule.c"
    Fortran 90 wrappers are saved to "/tmp/tmp0aQ2vk/src.linux-x86_64-2.7/libconfig-f2pywrappers2.f90"
  adding '/tmp/tmp0aQ2vk/src.linux-x86_64-2.7/fortranobject.c' to sources.
  adding '/tmp/tmp0aQ2vk/src.linux-x86_64-2.7' to include_dirs.
copying /build/usr/local/lib/python2.7/site-packages/numpy/f2py/src/fortranobject.c -> /tmp/tmp0aQ2vk/src.linux-x86_64-2.7
copying /build/usr/local/lib/python2.7/site-packages/numpy/f2py/src/fortranobject.h -> /tmp/tmp0aQ2vk/src.linux-x86_64-2.7
  adding '/tmp/tmp0aQ2vk/src.linux-x86_64-2.7/libconfig-f2pywrappers2.f90' to sources.
 - nothing done with h_files = ['config.h']
build_src: building npy-pkg config files
running build_ext
customize UnixCCompiler
customize UnixCCompiler using build_ext
customize Gnu95FCompiler
Found executable /usr/bin/gfortran
customize Gnu95FCompiler using build_ext
building 'libconfig' extension
compiling C sources
C compiler: gcc -pthread -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC

creating /tmp/tmp0aQ2vk/tmp
creating /tmp/tmp0aQ2vk/tmp/tmp0aQ2vk
creating /tmp/tmp0aQ2vk/tmp/tmp0aQ2vk/src.linux-x86_64-2.7
compile options: '-I. -I/build/dev_geoips/link/geoips/geoips/geoalgs/include -I/usr/include -I/tmp/tmp0aQ2vk/src.linux-x86_64-2.7 -I/build/usr/local/lib/python2.7/site-packages/numpy/core/include -I/build/usr/local/include/python2.7 -c'gcc: /tmp/tmp0aQ2vk/src.linux-x86_64-2.7/libconfigmodule.c
gcc: /tmp/tmp0aQ2vk/src.linux-x86_64-2.7/fortranobject.c
```

--- a/setup.py
+++ b/setup.py
@@ -368,13 +368,6 @@ def setup_package():
         cmdclass={"sdist": sdist_checked},
         python_requires='>=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*',
         zip_safe=False,
-        entry_points={
-            'console_scripts': [
-                'f2py = numpy.f2py.__main__:main',
-                'conv-template = numpy.distutils.conv_template:main',
-                'from-template = numpy.distutils.from_template:main',
-            ]
-        },
     )
 
     if "--force" in sys.argv:

